### PR TITLE
Release Flink 1.14.4

### PR DESCRIPTION
**Pull request by knaufk**

### CODE REVIEW

**Comment by sjwiesman** [comment]
Announcement page looks good

File: _config.yml [hunk 1]
```diff
@@ -218,7 +221,7 @@ release_archive:
       -
         version_short: "1.13"
         version_long: 1.13.6
-        release_date: 2022-02-09
+        release_date: 2022-02-18
```
Comment (Myasuka) [inline _config.yml:224]: If this PR is target for releasing 1.14, why we need to change the release date of 1.13?

File: _posts/2022-03-02-release-1.14.4.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+---
+
+The Apache Flink Community is pleased to announce another bug fix release for Flink 1.14.
+
+This release includes 51 bug and vulnerability fixes and minor improvements for Flink 1.14.
+Below you will find a list of all bugfixes and improvements (excluding improvements to the build infrastructure and build stability). For a complete list of all changes see:
+[JIRA](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12315522&version=12351074).
+
+We highly recommend all users to upgrade to Flink 1.13.4.
```
Comment (Myasuka) [inline _posts/2022-03-02-release-1.14.4.md]:
```suggestion
We highly recommend all users to upgrade to Flink 1.14.4.
```

**Review by Myasuka — COMMENTED** [review]
Thanks for creating this PR, I think we can mark it ready for review and merge instead of `draft` version.

File: _posts/2022-03-02-release-1.14.4.md (hunk 2 above)
Comment (knaufk) [reply _posts/2022-03-02-release-1.14.4.md]: good catch.

File: _config.yml (hunk 1 above)
Comment (knaufk) [reply _config.yml:224]: I corrected the release date in a separate commit: https://github.com/apache/flink-web/pull/510/commits/7bf169d8f3c7d76a1d53fd9d585b49e7f1899eda

**Review by MartijnVisser — APPROVED** [review]
Content looks good, LGTM

**Pull request merged**